File: app/src/main/java/de/cineaste/android/controllFlow/series/HistoryListSeriesTouchHelperCallback.java
package de.cineaste.android.controllFlow.series;


import android.content.res.Resources;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import de.cineaste.android.R;
import de.cineaste.android.adapter.series.SeriesListAdapter;
import de.cineaste.android.controllFlow.BaseSnackBar;

public class HistoryListSeriesTouchHelperCallback extends BaseSeriesTouchHelperCallback {

    public HistoryListSeriesTouchHelperCallback(Resources resources, LinearLayoutManager linearLayoutManager, RecyclerView recyclerView, SeriesListAdapter seriesListAdapter) {
        super(resources, linearLayoutManager, recyclerView, seriesListAdapter);
    }

    @Override
    protected int getIcon() {
        return R.drawable.ic_add_to_watchlist_white;
    }

    @Override
    protected BaseSnackBar getSnackBar() {
        return new SeriesSnackBarHistory(linearLayoutManager, recyclerView, seriesListAdapter);
    }

    @Override
    protected int getRightSwipeMessage() {
        return R.string.not_seen;
    }
}

File: app/src/main/kotlin/de/cineaste/android/controllFlow/series/HistoryListSeriesTouchHelperCallback.kt
package de.cineaste.android.controllFlow.series

import android.content.res.Resources
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import de.cineaste.android.R
import de.cineaste.android.adapter.series.SeriesListAdapter
import de.cineaste.android.controllFlow.BaseSnackBar

class HistoryListSeriesTouchHelperCallback(
    resources: Resources,
    linearLayoutManager: LinearLayoutManager,
    recyclerView: RecyclerView,
    seriesListAdapter: SeriesListAdapter
) : BaseSeriesTouchHelperCallback(resources, linearLayoutManager, recyclerView, seriesListAdapter) {

    override val icon: Int
        get() = R.drawable.ic_add_to_watchlist_white

    override val snackBar: BaseSnackBar
        get() = SeriesSnackBarHistory(linearLayoutManager, recyclerView, seriesListAdapter)

    override val rightSwipeMessage: Int
        get() = R.string.not_seen
}
